### PR TITLE
Woo Analytics: bring classes from module to package

### LINKED ISSUE #35630: Copy all code from the WooCommerce Analytics module to the new package
Epic: #35511

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #35630

## Proposed changes:

> [!IMPORTANT]
> Do not merge this PR until #35754 gets merged.

This brings in the codebase from the Jetpack module to the package. In the process:

- It adds new package dependencies instead of relying on Jetpack classes.
- It moves away from a singleton.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* Epic: #35511

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Not much to test just yet, the package isn't used anywhere. It will be easier to test via , which starts using the package in the Jetpack plugin.

### CODE REVIEW

File: projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php
Comment (darssen) [inline projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php:1]: Maybe change the file name to `class-woo-analytics-trait.php` ?

**Review by darssen — DISMISSED** [review]
Thanks for working on this @jeherve!
Approach looks good to me, moving the files to the new package. 

Since there is not much to test, I made sure to open all files in my IDE to check if there was some error regarding associations to the new classes and everything looked fine.

Also added a minor comment to the name of one class to follow standards.

File: projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php
Comment (jeherve) [reply projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php:1]: ah yeah, but then I should also update the class name to `Woo_Analytics_Trait`, right?

Comment (jeherve) [reply projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php:1]: I made the change in b207a51df4e9c1e53ba05757304006b898f13433

**Review by darssen — APPROVED** [review]
LGTM!

File: projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php
Comment (darssen) [reply projects/packages/woocommerce-analytics/src/class-wooanalyticstrait.php:1]: Thank you for taking care of it :)

**Pull request merged**